osu.Game/Screens/Play/HUD/GameplayLeaderboard.cs

```diff
@@ -29,32 +28,44 @@ public GameplayLeaderboard()
         /// <summary>
         /// Adds a player to the leaderboard.
         /// </summary>
-        /// <param name="currentScore">The bindable current score of the player.</param>
+        /// <param name="totalScore">A bindable of the player's total score.</param>
+        /// <param name="accuracy">A bindable of the player's accuracy.</param>
+        /// <param name="combo">A bindable of the player's current combo.</param>
         /// <param name="user">The player.</param>
-        public void AddPlayer([NotNull] BindableDouble currentScore, [NotNull] User user)
+        public void AddPlayer([NotNull] IBindableNumber<double> totalScore,
+                              [NotNull] IBindableNumber<double> accuracy,
+                              [NotNull] IBindableNumber<int> combo,
+                              [NotNull] User user)
         {
-            var scoreItem = addScore(currentScore.Value, user);
-            currentScore.ValueChanged += s => scoreItem.TotalScore = s.NewValue;
+            AddPlayer(totalScore, accuracy, combo, user, false);
         }
 
-        private GameplayLeaderboardScore addScore(double totalScore, User user)
+        /// <summary>
+        /// Adds a player to the leaderboard.
+        /// </summary>
+        /// <param name="totalScore">A bindable of the player's total score.</param>
+        /// <param name="accuracy">A bindable of the player's accuracy.</param>
+        /// <param name="combo">A bindable of the player's current combo.</param>
+        /// <param name="user">The player.</param>
+        /// <param name="localUser">Whether the provided <paramref name="user"/> is the local user.</param>
+        protected void AddPlayer([NotNull] IBindableNumber<double> totalScore,
+                                 [NotNull] IBindableNumber<double> accuracy,
+                                 [NotNull] IBindableNumber<int> combo,
+                                 [NotNull] User user, bool localUser) => Schedule(() =>
```

Comment:
does `localUser` have to get dragged through all these calls? `GameplayLeaderboardScore` already BDLs in an `IAPIProvider` - any reason not to use that and do an equality check there against `LocalUser`?

Reply:
Yeah, that was a misleading name, it's not `localUser`, but more of the `localPlayer`, as in the case of autoplay mods, the local player would be the autoplay's score "user".

Reply:
i don't understand how a `localUser` is different from a `localPlayer`. if it's related to autoplay (and only autoplay) then i'd just resolve the mods bindable and just check that.

Reply:
there's also the case of spectating, the player the user is spectating would be the `localPlayer` (though that makes the param name even more wrong, what do you think?)

Reply:
does that also apply to replays? if so i'd just go for `replayPlayer`.

Reply:
I've renamed it to `localOrReplayPlayer` for counting the local user in.